docs/architecture/gameStateUtil.md

```diff
@@ -0,0 +1,53 @@
+# GameStateUtil
+
+The [GameStateUtil](../../src/utils/gameStateUtil.ts) enforces a valid [GameState](../data/gameState.md)
+when loading and saving data via the [DatabaseUtil](./databaseUtil.md).
+
+## Load
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Actor
+  participant G as GameStateUtil
+  participant D as DatabaseUtil
+
+  A ->> G: load()
+  G ->> D: load()
+  D ->> G: data: unknown
+
+  G ->> G: validate(data)
+
+  alt data is GameState
+    G ->> G: gameState: GameState
+  else
+    G ->> G: init() <br> gameState: GameState
+    G ->> D: save(gameState)
+  end
+
+  G ->> A: gameState: GameState
+```
+
+## Save
```

Comment:
So currently `gameStateUtil` doesn't return anything. Should it return a `boolean`?

Reply:
I don't think so. The gameStateUtil's job is to load and save game data and handle any error states. So `gameStateUtil.save` is like a side effect function for now. I made a ticket somewhere to gracefully handle errors in the future so the Player knows when the game is in a bad state.

My idea for future error handling is to throw the error, have klickerKnight.ts catch, format, and output the error.